core/src/main/java/org/bitcoinj/core/MessageSerializer.java

```diff
@@ -39,9 +39,7 @@ public int getProtocolVersion() {
     }
 
     public void setProtocolVersion(int protocolVersion) {
-        this.protocolVersion = Math.min(protocolVersion, NetworkParameters.ProtocolVersion.CURRENT.getBitcoinProtocolVersion());
-        // the version we select should >= to MINIMUM
-        this.protocolVersion = Math.max(this.protocolVersion, NetworkParameters.ProtocolVersion.MINIMUM.getBitcoinProtocolVersion());
+        this.protocolVersion = protocolVersion;
```

Comment:
In the last PR a bug was found with min that should be max.

However, given that mainnet will be using a lower version of CURRENT protocol at first, this code would set the set the protocol version on messages to 70228 instead of 70227.  

There is probably still a bug here that we can't test for yet.

If we connect to a 70228 node as 70227, then that node will send us 70227 data, but we will interpret it as 70228, which for "mnlistdiff" should give an error.

Therefore, we should connect to nodes as 70228.  70227 nodes will send 70227 data and 70228 nodes send 70228 data.  

But this will probably cause some other problems later.  

It may be better to wait to release Dash Wallet when a significant number of nodes have upgraded to 19.2.

Reply:
I also discovered that a single serializer object is being used for all connected nodes, which means that setting the protocol version will cause problems if nodes use different protocol versions.

There were cases where messages were received from two nodes with two protocol versions, both were setting the default single serializer protocol version when the message was received this resulted in a race condition where a 70228 nodes messages were being processed as 70227.

Therefore, for now, we won't set the protocol version until a new serializer is created for each node based on the protocol version.  This will be handled in a new story.